src/hooks/api/useBuildGetter.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react'
+import useFetch from 'hooks/useFetch'
+import { GetHookResponse } from './types'
+
+export const useBuildGetter = <T, U extends Array<unknown> = Array<any>>(
+  paramBuilder: (...args: U) => [string, Record<string, any>]
+): GetHookResponse<T, U> => {
+  const { data, get: genericGet, isLoading, error, isSuccess } = useFetch<T>()
+  const get = useCallback(
+    (...args: U) => genericGet(...paramBuilder(...args)),
+    []
```

Comment:
missing dependencies

Reply:
this is intentional, to make sure this callback never changes